src/app/component/SliderWidget/SliderWidget.component.js

```diff
@@ -45,12 +46,54 @@ export class SliderWidget extends PureComponent {
         slider: [{}]
     };
 
-    state = { activeImage: 0 };
+    state = {
+        activeImage: 0,
+        carouselDirection: 'right',
+        imageToShow: 0
+    };
+
+    componentDidUpdate(prevProps) {
+        const { slider: { slideSpeed } } = this.props;
+        const { slider: { slideSpeed: prevSlideSpeed } } = prevProps;
+
+        this.getImageToShow();
+
+        if (slideSpeed !== prevSlideSpeed) {
+            this.startCarousel(slideSpeed);
+        }
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.carouselInterval);
+    }
 
     onActiveImageChange = (activeImage) => {
         this.setState({ activeImage });
     };
 
+    startCarousel = (interval) => {
+        this.carouselInterval = setInterval(() => {
+            const { imageToShow } = this.state;
+
+            this.onActiveImageChange(imageToShow);
+        }, interval);
+    };
+
+    getImageToShow() {
+        const { activeImage, carouselDirection } = this.state;
+        const { slider: { slides } } = this.props;
+
+        if (activeImage === 0) {
+            this.setState({ carouselDirection: 'right' });
+            this.setState({ imageToShow: slides.length - 1 !== 0 ? activeImage + 1 : activeImage });
+        } else if (activeImage === slides.length - 1) {
+            this.setState({ carouselDirection: 'left' });
+            this.setState({ imageToShow: activeImage - 1 });
```

Comment:
Avoid calling multiple setStates in a single update cycle, they may overwrite each other. Use callback-based `setState(() => ({ ... }))` when you indeed need multiple setStates in a single update cycle, but in this specific case you don't. Just unify the setStates, this will be sufficient.